spec/integ/crypto/verification.spec.ts

```diff
@@ -990,6 +992,31 @@ describe.each(Object.entries(CRYPTO_BACKENDS))("verification (%s)", (backend: st
             aliceClient.setGlobalErrorOnUnknownDevices(false);
             syncResponder.sendOrQueueSyncResponse(getSyncResponse([BOB_TEST_USER_ID]));
             await syncPromise(aliceClient);
+            const crypto = aliceClient.getCrypto()!;
+            if (crypto instanceof RustCrypto) {
```

Comment:
I *really* don't like that we're using `instanceof` for this - apart from breaking OO paradigms, it has a tendency to break in unpredictable ways if your modules get messed up and you end up with two different `RustCrypto` classes. If the app really needs to know what it is, an `isRustCrypto()`? Or `implNeedsSenderDeviceKeys()` (although maybe that's even worse).

Reply:
The right solution here is probably a test on `backend`, as per https://github.com/uhoreg/matrix-js-sdk/blob/crypto-wasm-11/spec/integ/crypto/verification.spec.ts#L125

Reply:
Yes, I originally tested it using `backend`, but then TypeScript would complain because I'm calling functions that are only defined for `RustCrypto` (`onSyncCompleted` and `preprocessToDeviceMessages`)

Reply:
ah, of course. I guess cast it... or test for the existence of those functions?

Reply:
I've changed it to check using `backend`, and then cast the `crypto` object to make TypeScript happy